project_report/project_information.py

```diff
@@ -53,300 +48,230 @@ def __init__(self, project_name):
+            bcl2fastq_version = None
+            if os.path.isfile(prog_vers_yaml):
+                with open(prog_vers_yaml, 'r') as open_file:
+                    full_yaml = yaml.safe_load(open_file)
+                    if 'bcl2fastq' in full_yaml and full_yaml.get('bcl2fastq'):
+                        bcl2fastq_version = full_yaml.get('bcl2fastq')
+            if bcl2fastq_version:
+                bcl2fastq_versions.add(bcl2fastq_version)
+            else:
+                raise ValueError('Run %s has no bcl2fastq version available in %s' % (run_id, prog_vers_yaml))
+        return ', '.join(bcl2fastq_versions)
 
-    def get_lims_sample(self, sample_name):
-        samples = [s for s in self.samples_for_project_lims if s.name == sample_name]
-        if len(samples) == 1:
-            return samples[0]
-        raise ValueError('%s samples found for %s' % (len(samples), sample_name))
+    @staticmethod
+    def _read_program_version_yaml(prog_vers_yaml):
+        program_version = {}
+        with open(prog_vers_yaml, 'r') as open_file:
+            full_yaml = yaml.safe_load(open_file)
+            for p in ['bwa', 'gatk', 'samtools', 'samblaster', 'biobambam_sortmapdup']:
+                if p in full_yaml:
+                    program_version[p + '_version'] = full_yaml.get(p)
+        return program_version
+
+    def _get_version(self, sample_name):
+        versions = {}
+        genome_version = None
+        species = self.get_species_from_sample(sample_name)
+        sample_source = path.join(self.project_source, sample_name)
+        if species == 'Homo sapiens':
+            program_csv = find_file(sample_source, 'programs.txt')
+            versions.update(self._read_program_csv(program_csv))
+            summary_yaml = find_file(sample_source, 'project-summary.yaml')
+            if summary_yaml:
+                bcbio_version, genome_version = self._read_project_summary_yaml(summary_yaml)
+                versions['bcbio_version'] = bcbio_version
+        else:
+            program_yaml = find_file(sample_source, 'program_versions.yaml')
+            versions.update(self._read_program_version_yaml(program_yaml))
+        versions['bcl2fastq_version'] = self._get_bcl2fastq_version(
+            query_dict(self.sample_info(sample_name), 'data.aggregated.run_ids')
+        )
+        if 'biobambam_sortmapdup_version' in versions:
+            versions['biobambam_or_samblaster'] = 'biobambam'
+            versions['biobambam_or_samblaster_version'] = versions.get('biobambam_sortmapdup_version')
+        else:
+            versions['biobambam_or_samblaster'] = 'samblaster'
+            versions['biobambam_or_samblaster_version'] = versions.get('samblaster_version')
+        if not genome_version:
+            genome_version = self._get_genome_version(sample_name)
+        versions['genome_version'] = genome_version
+        return versions
+
+    # Sample spcecific functions
+    def sample_info(self, sample_id):
```

Comment:
This the main function gathering sample information. All the data is retrieved once through various queries and parsing and cached.